### PR TITLE
Multipolygon support

### LINKED ISSUE #143: Add support for Multipolylines
I load saved polylines from other drawing sessions. Can't figure out how to add them to PM to be editable and snappable. Is there a way to do this?

### LINKED ISSUE #249: Uncaught TypeError: Cannot read property 'lat' of null
I'm not sure if this error is due to leaflet.pm or is leaflet itself that has the bug, but certainly it didn't appear until I added leaflet.pm to my package.

The error appears when you add a polyline that has an array of arrays of [lat,lng], which leds me to believe it has something to do with the lack of support to MultiPolylines in leaflet.pm itself.

[Fiddle]( http://jsfiddle.net/t5s1yn3z/)

### PR DESCRIPTION
**Pull request by codeofsumit**
this is an open PR that is not 100% finished yet. But I'm working on Multipolygon & Multipolyline support.
I'd like to make support as robust as possible without hard coded array depths or flattening.
Wish me luck 🤞 

Will fix #143, fix #249, close #279, close #293 

UPDATE:
This PR is quite far: I'd love to have some developers test it with their apps and data.

### CODE REVIEW

**Comment by codeofsumit** [comment]
current status:
multipolygons can be added to leaflet.pm, vertexes edited, added and removed 🎉

**Comment by codeofsumit** [comment]
multipoly

**Comment by codeofsumit** [comment]
Cutting with MultiPolygon Support

cut multypoly

**Comment by codeofsumit** [comment]
@mpschaeuble, @smeijer, I'd love to get some feedback on this. Could you test this branch with your data and apps for MultiPolygon and MultiPolyline support?

File: src/js/Edit/L.PM.Edit.Line.js [hunk 1]
```diff
@@ -349,8 +352,11 @@ Edit.Line = Edit.extend({
 
         // if the ring of the poly has no coordinates left, remove the ring
         if (coordsRing.length <= 1) {
+            // find the parent array index path of the coords ring
+            const coordsRingParentIndexPath = parentPath.slice(0, indexPath.length - 1);
+
             // remove coords ring
-            coords.splice(ringIndex, 1);
+            get(coords, coordsRingParentIndexPath).splice(parentPath, 1);
```
Comment (codeofsumit) [inline src/js/Edit/L.PM.Edit.Line.js]: This probably won’t work for polylines or regular polygons?

**Comment by smeijer** [comment]
Sorry for the late response.

The latest version with this PR merged seems to be working just fine. Awesome work!

Thanks. 🎉

**Comment by adamsjoe96** [comment]
i have the same issue with the multipolygon when i load the ressources from my server then transform the ressources in geojson format then try to edit by clicking on the multipolygon in the map

**Comment by codeofsumit** [comment]
@adamsjoe96 please open a new issue and describe exactly the problem you have.
From your comment here it looks like you want to edit your polygon by clicking on it? If so, you have to add a click event to the polygon and enable edit in the click handler.

**Pull request merged**